apps/meteor/server/services/upload/service.ts

```diff
@@ -22,12 +22,31 @@ export class UploadService extends ServiceClassInternal implements IUploadServic
 		return sendFileLivechatMessage({ roomId, visitorToken, file, msgData: message });
 	}
 
-	async getFileBuffer({ file }: { userId: string; file: IUpload }): Promise<Buffer> {
+	async getFileBuffer({ file }: { file: IUpload }): Promise<Buffer> {
 		const buffer = await FileUpload.getBuffer(file);
 
 		if (!(buffer instanceof Buffer)) {
 			throw new Error('Unknown error');
 		}
 		return buffer;
 	}
+
+	async extractMetadata(file: IUpload): Promise<{ height?: number; width?: number; format?: string }> {
+		return FileUpload.extractMetadata(file);
+	}
+
+	async parseFileIntoMessageAttachments(
+		file: Partial<IUpload>,
+		roomId: string,
+		user: IUser,
+	): Promise<{
+		files: {
+			_id: string;
+			name: string | undefined;
+			type: string | undefined;
+		}[];
+		attachments: MessageAttachment[];
+	}> {
```

Comment:
Not sure what's better, typing this again, or type it as `ReturnType<parseFileIntoMessageAttachments>` now that's properly typed